src/main/com/fulcrologic/rad/database_adapters/datomic_options.cljc

```diff
@@ -30,3 +30,20 @@
 
    This only affects Datomic autogenerated resolvers."
   :com.fulcrologic.rad.database-adapters.datomic/wrap-resolve)
+
+(def raw-txn
+  "Pathom enviroment key. A sequence of Datomic transaction data in the list and/or map form, which will be added to the
+   transaction data produced by the currently active (save or delete) middleware.
+   
+   You can use this to ensure pre- and post-condition, enrich the transactions, etc.
+   
+   Ex.:
+   ```clj
+   (let [user ..., order ..., ident ...]
+    (cond-> pathom-env
+      (first-order-campaign? order)
+      (assoc do/raw-txn [(list 'my.txn.fns/ensure-no-prior-order user) ; check pre-condition
```

Comment:
This is not really meant to go in at the *pathom* layer (at least not primarily). It is meant to be installed as part of the middleware stack. To that end, perhaps it makes sense to add a datomic-common helper function like `(append-to-raw-transaction mw-env) -> updated-env`?

I mean, I guess it is *ok* to put it in the pathom layer...in either case, though, the implementation should always assume others might be using it in middleware and should be appending to what is already there:

```
(update env do/raw-txn (fnil into []) my-txn)
```